src/core/DataFile.cpp

```diff
@@ -163,6 +163,53 @@ DataFile::~DataFile()
 
 
 
+bool DataFile::validate( QString extension )
+{
+	switch( m_type )
+	{
+	case Type::SongProject:
+		if( extension == "mmp" || extension == "mmpz" )
+		{
+			return true;
+		}
+		break;
+	case Type::SongProjectTemplate:
+		if(  extension == "mpt" )
+		{
+			return true;
+		}
+		break;
+	case Type::InstrumentTrackSettings:
+		if ( extension == "xpf" || extension == "xml" )
+		{
+			return true;
+		}
+		break;
+	case Type::UnknownType:
+		if (! ( extension == "mmp" || extension == "mpt" || extension == "mmpz" ||
```

Comment:
At second glance Dave's logic looks OK.

It appears we register our plugins as having an extension associated... and I'm guessing the `Engine::pluginFileHandling().contains()` doesn't actually contain `xiz` for some reason.  I'd guess this is actually an existing bug that surfaced with this code... but that is just a hunch.... still investigating...